domain/relation/state/relation_test.go

```diff
@@ -347,6 +348,71 @@ func (s *relationSuite) TestGetRelationEndpointsRelationNotFound(c *gc.C) {
 	c.Assert(err, jc.ErrorIs, relationerrors.RelationNotFound)
 }
 
+func (s *relationSuite) TestGetRelationDetails(c *gc.C) {
+	// Arrange: Add two endpoints and a relation on them.
+	relationUUID := "fake-relation-uuid"
+	relationID := 7
+
+	charmRelationUUID1 := "fake-charm-relation-uuid-1"
+	applicationEndpointUUID1 := "fake-application-endpoint-uuid-1"
+	relationEndpointUUID1 := "fake-relation-endpoint-uuid-1"
```

Comment:
Use GenRelationUUID(c), etc. 

After 3 PRs of the same request, please just do it rather doing it later in a new PR.

Reply:
Changed in all 3.

Reply:
You can use `uuid.MustNewUUID().String()` for the rest.

Changing them to strings is starting to not work. You frequently have to make them type again to use in method under test. The string could be handled in the add methods.